### DISCUSSION TITLE
paperless-ngx cannot update

### DISCUSSION THREAD

**Discussion by f1n3d** [tteck/Proxmox#2622] [Help Needed] [answered]
Hi,
I tried to update my paperless-nix on version 2.0.0 container by the "update" command, but it runs into an error:
<img width="974" alt="Pasted Graphic" src="https://github.com/tteck/Proxmox/assets/93422567/56f4ac53-fb38-4d8a-8ce9-63e9c01ec363">
How can I fix it?
Thanks in advance for your help.

**Comment by manuelkue** [comment]
Have you checked if your lxc container has some memory left?
I once had a very similar error on updating and I just haven't seen that it ran out of memory 🙈
If that's the case just assign it some more.

**Reply by f1n3d** [reply]
Hi manuelkue,

my container has 2gb ram. Ive tried the update with 4gb again, but I ran into the same error.

**Reply by manuelkue** [reply]
Ah sorry, I meant disk space, not RAM.

**Reply by f1n3d** [reply]
The Container has 5GB of 32GB Disk Space in use. So it should be enough, or am I wrong?

**Reply by manuelkue** [reply]
Ok, then it's another issue I suppose 🤷

**Comment by tteck** [comment, marked as answer]
Have you attempted updating line by line manually to identify the specific issue hindering the completion of the update?

Since I don't personally utilize Paperless-ngx, it's challenging for me to troubleshoot your problem. However, when I perform a fresh installation and proceed with the update process, I encounter no issues.
```
    ____                        __                                     
   / __ \____ _____  ___  _____/ /__  __________    ____  ____ __  __
  / /_/ / __ `/ __ \/ _ \/ ___/ / _ \/ ___/ ___/___/ __ \/ __ `/ |/_/
 / ____/ /_/ / /_/ /  __/ /  / /  __(__  |__  )___/ / / / /_/ />  <  
/_/    \__,_/ .___/\___/_/  /_/\___/____/____/   /_/ /_/\__, /_/|_|  
           /_/                                         /____/        
 
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 1
Using Root Password: Automatic Login
Container ID: 110
Using Hostname: paperless-ngx
Using Disk Size: 10
Allocated Cores: 2
Allocated RAM: 2048
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Using APT-Cacher IP Address: Default
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Paperless-ngx LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using thin-pool for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 110 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.250 
 ✓ IPv4 Internet Connected
 ✗ IPv6 Internet Not Connected
 ✓ DNS Resolved github.com to 140.82.112.4
 ✓ Updated Container OS
 ✓ Installed Python3
 ✓ Installed Dependencies
 ✓ Installed OCR Dependencies
 ✓ Installed JBIG2
 ✓ Installed Paperless-ngx
 ✓ Installed Natural Language Toolkit
 ✓ Set up PostgreSQL database
Would you like to add Adminer? <y/N> 
 ✓ Set up admin Paperless-ngx User & Password
 ✓ Created Services
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!

Paperless-ngx should be reachable by going to the following URL.
         http://192.168.86.250:8000 
```

```
    ____                        __                                     
   / __ \____ _____  ___  _____/ /__  __________    ____  ____ __  __
  / /_/ / __ `/ __ \/ _ \/ ___/ / _ \/ ___/ ___/___/ __ \/ __ `/ |/_/
 / ____/ /_/ / /_/ /  __/ /  / /  __(__  |__  )___/ / / / /_/ />  <  
/_/    \__,_/ .___/\___/_/  /_/\___/____/____/   /_/ /_/\__, /_/|_|  
           /_/                                         /____/        
 
 ✓ Stopped Paperless-ngx
 ✓ paperless-task-queue.service Exists.
 ✓ Updated to v2.6.2
 ✓ Cleaned
 ✓ Started Paperless-ngx
 ✓ Updated Successfully!

root@paperless-ngx:~# 
```
Maybe try 
```
cd /opt/paperless/src
python3 manage.py migrate
```
To see if it reveals anything. 🤷🏻‍♂️

**Reply by f1n3d** [reply]
Thank you for your help. The Command showed that the specific Paperless URL setting in the paperless.conf file caused the error. after setting a # in front of the two lines "PAPERLESS_URL" and "PAPERLESS_CSRF_TRUSTED_ORIGINS" the update worked like a charm. :)
after the update I removed the # and everything works again. 
Thank you for your great work btw.

**Reply by tteck** [reply]
Were those settings initially commented out? I don't see any part in the script where it deviates from the [configuration example]( https://raw.githubusercontent.com/paperless-ngx/paperless-ngx/main/paperless.conf.example) file that's being imported.
```
#PAPERLESS_URL=https://example.com
#PAPERLESS_CSRF_TRUSTED_ORIGINS=https://example.com # can be set using PAPERLESS_URL
```

**Reply by f1n3d** [reply]
Initially those settings were commented out. I added my url config there and removed the #.
That caused the issue while upgrading.
